pennylane/estimator/templates/select.py

```diff
@@ -38,19 +38,24 @@ class SelectTHC(ResourceOperator):
 
             This decomposition assumes that an appropriately sized phase gradient state is available.
             Users should ensure that the cost of constructing this state has been accounted for.
-            See also :class:`~.pennylane.estimator.templates.PhaseGradient`.
+            See also :class:`~.pennylane.estimator.templates.subroutines.PhaseGradient`.
 
     Args:
         thc_ham (:class:`~pennylane.estimator.compact_hamiltonian.THCHamiltonian`): A tensor hypercontracted
-            Hamiltonian on which the select operator is being applied.
+            Hamiltonian on which this ``Select`` operator is being applied.
+        num_batches (int): The maximum number of batches for loading the rotation angles
+            into temporary quantum registers for Givens rotations.
+            The default value of ``1`` loads all angles in one batch.
```

Comment:
```suggestion
        num_batches (int): The number of batches for loading the rotation angles
            into temporary quantum registers for Givens rotations.
            The default value of ``1`` loads all angles in one batch.
```

And everywhere.

Maybe mention that inverse relationship between num_batches and num_qubits.